06.bowling_object/frame.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+require_relative 'shot'
+
+class Frame
+  MAX_SHOTS_IN_LAST_FRAME = 3
+  MAX_SHOTS_IN_REGULAR_FRAME = 2
+
+  attr_reader :scores, :frame_idx
+
+  def initialize(frame_idx)
+    @scores = []
+    @frame_idx = frame_idx
+  end
+
+  def add_shot(marks, score_idx)
+    loop do
+      @scores << Shot.new(marks[score_idx])
+      score_idx += 1
+      break if complete?
+    end
+    score_idx
+  end
+
+  def score(frames)
+    @scores.sum(&:score) + bonus_score(frames)
+  end
+
+  private
+
+  def complete?
+    if @frame_idx != Game::LAST_FRAME
+      strike? || @scores.size == MAX_SHOTS_IN_REGULAR_FRAME
+    else
+      @scores.size == MAX_SHOTS_IN_LAST_FRAME || (@scores.size == MAX_SHOTS_IN_REGULAR_FRAME && !strike? && !spare?)
```

Comment:
```suggestion
      @scores.size == (strike? || spare? ? MAX_SHOTS_IN_LAST_FRAME : MAX_SHOTS_IN_REGULAR_FRAME)
```

若干トリッキーですが、こんな書き方もできそうです

Reply:
この書き方であれば、コードを短縮できますね。